lib/segment/src/segment.rs

```diff
@@ -560,6 +560,21 @@ impl SegmentEntry for Segment {
     fn check_error(&self) -> Option<SegmentFailedState> {
         self.error_status.clone()
     }
+
+    fn delete_filtered<'a>(
+        &'a mut self,
+        op_num: SeqNumberType,
+        filter: &'a Filter,
+    ) -> OperationResult<Vec<PointIdType>> {
+        let mut deleted_points = Vec::new();
```

Comment:
Why do we need to collect deleted ids here?

Reply:
Well, ne immediate reason right now... Tell me what you want ... I am too unexperienced with the project to decide wether to return the number of deleted points or the deleted point ids. Or maybe to trace / log the deleted point ids but not return them... depends on how deep you want to analyse this. at least for debugging i find it quite useful to have the most information possible.

Reply:
> I am too unexperienced with the project to decide wether to return the number of deleted points or the deleted point ids.

Sure, absolutely no problem with that ;) I was just not sure if this is intentional or not. If not, I think it would be enough to just return a number of removed points in this specific place.

Thank you for your patience in fixing my nitpicks!

Reply:
No Problem! Just patched in your latest requests. I did not add any tests so far and it doesn't really make sense since I cannot compile the project locally anymore (cargo check and cargo clippy works fine)